Firmware/Main/CAN.cpp

```diff
@@ -208,21 +220,31 @@ void requestCellVoltages() {
     next_can_id = BMSC1_LTC1_REQUEST_CELLS;
 }
 
-void canTask(void *canData) {
-  TickType_t last_request = xTaskGetTickCount();
-  int requests = 0;
+void canSendTask(void *canData) {
   while (1) {
     /* NOTE: CAN breaks if we try sending messages with 0 other nodes on the bus.
     / Therefore, change CAN_NODES in Main.h to make sure things dont break. */
     if (CAN_NODES != 0) {
+      requestCellVoltages();
+    }
+    /* Ask for other half of cell voltages from BMS every 2 seconds, test timings later to improve boot performance */
+    vTaskDelay(2 * configTICK_RATE_HZ);
+  }
+}
+
+void canReceiveTask(void *canData) {
+  messageQueue = xQueueCreate(16, sizeof(queue_message));
+  if (messageQueue == NULL) {
+    Serial.println("Failed to create queue");
+    while (1) {
+      vTaskDelay((50 * configTICK_RATE_HZ) / 1000);
+    }
+  }
+  CAN_bus.enableFIFOInterrupt();
+  CAN_bus.onReceive(addToQueue);
```

Comment:
LGTM. Have you had a chance to test?

Reply:
I only tested it with another Teensy sending messages every 1 ms but it seemed to work fine, and it sends messages closer to once every two seconds than before.